packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/Credentials.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb
+
+import io.realm.interop.NativePointer
+import io.realm.interop.RealmInterop
+
+/**
+ * TODO
+ */
+sealed class Credentials(val id: String) {
```

Comment:
This looks like it need a bigger refactor. I tried to migrate the Java API's directly here: https://github.com/realm/realm-kotlin/pull/437/files#diff-1052d1ac32f6e72c1fccfefcf6262a978f35df4e663d72b7a16da9d7708355bdR54

But I can see some advantage to using sealed classes.

But we should add a TODO about revisiting the API here.

Reply:
Added a TODO referencing this comment